### DISCUSSION TITLE
error or bug when create custom resource page

### DISCUSSION THREAD

**Discussion by Darnes1958** [filamentphp/filament#13398] [Help]
### Package

Other

### Package Version

v3

### How can we help you?

after create custom page (MainCreate)  in (MainResource) and put a form in it  and I call it with url( 'mains/maincreate') in ListPage . database error occur . (SELECT top 1 * FROM [mains] WHERE [id] = maincreate) .

**Comment by youridijk** [comment]
I encountered almost the same error. In my case I created a view page for a resource and added it manually to the list of routes for the resource. The problem was the order of the routes that caused the error:

Wrong orders (note "view" and "create")
```php
public static function getPages(): array
{
    return [
        'index' => Pages\ListUsers::route('/'),
        'view' => Pages\ViewUser::route('/{record}'),
        'create' => Pages\CreateUser::route('/create'),
        'edit' => Pages\EditUser::route('/{record}/edit'),
    ];
}
```

Right order (again, note "view" and "create")
```php
public static function getPages(): array
{
    return [
        'index' => Pages\ListUsers::route('/'),
        'create' => Pages\CreateUser::route('/create'),
        'view' => Pages\ViewUser::route('/{record}'),
        'edit' => Pages\EditUser::route('/{record}/edit'),
    ];
}
```

You need to make sure that "create" is **before** view, otherwise, Filament things you want to view a record with ID "create". This isn't handled in the code, so it will crash in the DB.
